dex/elrond_dex_pair/src/lib.rs

```diff
@@ -475,14 +495,33 @@ pub trait Pair:
         let old_k = self.calculate_k_for_reserves();
 
         let mut reserve_token_out = self.pair_reserve(&token_out).get();
+        let mut virtual_reserve_token_out = self.pair_virtual_reserve(&token_out).get();
         require!(
             reserve_token_out > amount_out_min,
             "Insufficient reserve for token out"
         );
+        require!(
```

Comment:
I would move this check above the check with reserve_token_out. From my understanding, virtual_reserve_token_out should be always equal or grater than the reserve_token_out. So if virtual_reserve_token_out < amount_out_min then it should be clear that you don't have enough liquidity to perform this swap in any of the siblings contracts